### PR TITLE
test(swapName):add solution & test

### PR DESCRIPTION
**Pull request by Hamzaelkhatri**

### CODE REVIEW

**Review by zainabdnaya — APPROVED** [review]
All Good

**Review by davidrobert99 — COMMENTED** [review]
It is a good idea to improve the code. There are a lot of "for" cycles.

File: tests/swapname_test/main.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"github.com/01-edu/go-tests/lib/challenge"
+)
+
+func main() {
+
+	args := [][]string{
+		{"John Doe"},
+		{"Hello World"},
```
Comment (MSilva95) [inline tests/swapname_test/main.go:11]: Same here in the tests, we should use names as well

File: solutions/swapname/main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package main
```
Comment (MSilva95) [inline solutions/swapname/main.go:1]: Try to create a solution not using (make) and (strings.TrimSpace) so that we can restrict this functions.

**Review by MSilva95 — COMMENTED** [review]
If You cant create a solution without the functions mentioned in the comments, we need to recheck if this exercise will be in the exams or not

File: solutions/swapname/main.go (hunk 2 above)
Comment (Hamzaelkhatri) [reply solutions/swapname/main.go:1]: i changed , ty

**Review by MSilva95 — CHANGES_REQUESTED** [review]

File: solutions/swapname/main.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	return str[j : i+1]
+}
+
+func main() {
+	argument := os.Args[1:]
+	if len(argument) == 1 {
+		fullName := isValidName(splitWhiteSpace(splitWhiteSpace(argument[0])))
+		if fullName != nil && len(fullName) == 2 {
+			fmt.Println("[" + fullName[1] + " " + fullName[0] + "]")
```
[the remark this answers is not in the document]
Comment (Hamzaelkhatri) [reply solutions/swapname/main.go]: fixed

**Review by MSilva95 — CHANGES_REQUESTED** [review]
add one valid test with extra spaces in the middle of the arg

**Review by MSilva95 — APPROVED** [review]

**Pull request merged**